lib/overcommit/git_repo.rb

```diff
@@ -54,16 +54,24 @@ def modified_files(options)
         map { |relative_file| File.expand_path(relative_file) }
     end
 
-    # Returns the names of all files that are tracked by git.
+    # Returns the names of files in the given paths that are tracked by git.
     #
+    # @param paths [Array<String>] variable-length list of paths to check
     # @return [Array<String>] list of absolute file paths
-    def all_files
-      `git ls-files`.
+    def list_files(*paths)
```

Comment:
I don't think there is a need to make this a splat? Instead it can just default to the empty array. As long as it's an `Enumerable` we don't need to care.

Having to remember whether a method takes a list of arguments or just a list, I would prefer it always took the latter.